runtime/src/main/java/org/corfudb/runtime/ViewsMaintenance.java

```diff
@@ -13,49 +14,90 @@
- * parts of the log.
+ * parts of the log. Also, auto sync objects to amortize the log sync costs.
  * <p>
  * Created by Maithem on 11/16/18.
  */
 
 @Slf4j
 @NotThreadSafe
-public class ViewsGarbageCollector {
+public class ViewsMaintenance {
 
 
     private long trimMark = Address.NON_ADDRESS;
 
     @Getter
     private boolean started;
 
-    final ScheduledExecutorService gcThread = Executors.newSingleThreadScheduledExecutor(
+    /**
+     * How often to sync opened objects
+     */
+    @Getter
+    @Setter
+    long objectSyncDelayInSeconds = 30;
+
+    /**
+     * How often to run GC on the opened objects
+     */
+    @Getter
+    @Setter
+    long objectGCPeriodInSeconds = 20 * 60;
+
+    /**
+     * This single thread is shared by the two tasks (i.e. object GC and object syncing)
+     */
+    final ScheduledExecutorService maintenance = Executors.newSingleThreadScheduledExecutor(
             new ThreadFactoryBuilder().setDaemon(true)
-                    .setNameFormat("ViewsGarbageCollector")
+                    .setNameFormat("ViewsMaintenance")
                     .build());
 
     final CorfuRuntime runtime;
 
-    public ViewsGarbageCollector(CorfuRuntime runtime) {
+    public ViewsMaintenance(CorfuRuntime runtime) {
         this.runtime = runtime;
         this.started = false;
     }
 
     public void start() {
-        gcThread.scheduleAtFixedRate(() -> runRuntimeGC(),
-                runtime.getParameters().getRuntimeGCPeriod().toMinutes(),
-                runtime.getParameters().getRuntimeGCPeriod().toMinutes(),
-                TimeUnit.MINUTES);
+        maintenance.scheduleAtFixedRate(() -> runRuntimeGC(),
+                objectGCPeriodInSeconds, objectGCPeriodInSeconds,
+                TimeUnit.SECONDS);
+        maintenance.scheduleAtFixedRate(() -> runSyncObjects(),
+                objectSyncDelayInSeconds, objectGCPeriodInSeconds,
+                TimeUnit.SECONDS);
         this.started = true;
     }
 
     public void stop() {
-        gcThread.shutdownNow();
+        maintenance.shutdownNow();
         this.started = false;
     }
 
+    /**
+     * Objects are synced on access, if a client access an object irregularly, then over time
+     * it can experience unpredictable latencies because of syncing. When used, the ViewsMaintinance
+     * will periodically sync the object ao amortize the sync costs.
+     */
+    private void runSyncObjects() {
+        try {
+            long s1 = System.currentTimeMillis();
+            runtime.getObjectsView().syncObjects();
+            long s2 = System.currentTimeMillis();
+            log.info("syncObjects: sync took {} ms, total maps {}", s2 - s1, runtime.getObjectsView()
+                    .getObjectCache().size());
+        } catch (Exception e) {
+            if (e.getCause() instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+                throw new GarbageCollectorException(e);
```

Comment:
Is this the right error to throw? It is not a GC error, should we introduce a SyncObjectException or viewsMaintenanceException?